File: backend/src/routes/health.routes.ts
import { Router } from 'express'
import { requireAuth } from '../middlewares/auth.middleware'
import { listarLogs, upsertLog } from '../controllers/health.controller'

const router = Router()

router.use(requireAuth)

router.get('/',         listarLogs) // ?since=YYYY-MM-DD
router.put('/:date',    upsertLog)  // upsert do dia (YYYY-MM-DD)

export default router

File: backend/internal/health/health_routes.go
package health

import (
	"github.com/gin-gonic/gin"
	"github.com/jackc/pgx/v5/pgxpool"

	"beeplanerofc/backend-go/internal/middleware"
	"beeplanerofc/backend-go/internal/supabaseauth"
)

// RegisterRoutes registra as rotas de saúde no grupo /api/health.
func RegisterRoutes(router *gin.RouterGroup, db *pgxpool.Pool, authClient *supabaseauth.Client) {
	ctrl := NewController(db)

	// Rota pública — sync via iOS Shortcut (autenticação por sync_token no body)
	router.POST("/sync", ctrl.SyncFromShortcut)

	// Rotas autenticadas
	authed := router.Group("")
	authed.Use(middleware.RequireAuth(db, authClient))
	authed.GET("", ctrl.ListarLogs)
	authed.PUT("/:date", ctrl.UpsertLog)
	authed.GET("/sync-token", ctrl.GetSyncToken)
}
